main.go

```diff
@@ -68,6 +68,8 @@ import (
 //// 	@scope.admin							Grants read and write access to administrative information
 
 func main() {
+	localCtx, localCancel := context.WithCancel(context.Background())
```

Comment:
I don't think we are on golang 1.22 yet, but maybe add a TODO for switching to https://pkg.go.dev/context#AfterFunc when we get there?

Reply:
I'm not sure in retrospect whether we could have just done this with a normal channel as well.  Missed this when merging, but I'm getting a bit concerned about the number of TODOs we're adding but never resolving.  Might be best to add things to tickets if we think they need doing just so they're visible.